internal/contentmanager/contentmanager.go

```diff
@@ -54,23 +61,46 @@ func (i *instance) ManageContent(ctx context.Context, ctrl controller.Controller
 
 	// TODO: add a http.RoundTripper to the config to ensure it is always using an up
 	// to date authentication token for the ServiceAccount token provided in the ClusterExtension.
-	// Maybe this should be handled by the RestConfigMapper?
+	// Maybe this should be handled by the RestConfigMapper
+
+	// Assumptions: all objects received by the function will have the Object metadata specfically,
+	// ApiVersion and Kind set. Failure to which the code will panic when adding the types to the scheme
+	scheme := runtime.NewScheme()
+	for _, obj := range objs {
+		gvk := obj.GetObjectKind().GetVersionKind()
```

Comment:
Ref: https://pkg.go.dev/k8s.io/apimachinery/pkg/runtime/schema#ObjectKind

```suggestion
		gvk := obj.GetObjectKind().GroupVersionKind()
```

Applies to all other places the same error is popping up